src/Crypto/EncryptionKey.php

```diff
@@ -0,0 +1,39 @@
+<?php declare(strict_types = 1);
+
+namespace Cmnty\Push\Crypto;
+
+use InvalidArgumentException;
+
+class EncryptionKey implements KeyingMaterial
+{
+    /**
+     * @var BinaryString
+     */
+    private $binaryString;
+
+    /**
+     * Create encryption key.
+     *
+     * @param BinaryString $binaryString
+     *
+     * @throws InvalidArgumentException When the encryption key is not the correct length.
+     */
+    public function __construct(BinaryString $binaryString)
+    {
+        if ($binaryString->getLength() != 16) {
+            throw new InvalidArgumentException('EncryptionKey could not be created: incorrect length.');
+        }
+
+        $this->binaryString = $binaryString;
+    }
+
+    /**
+     * Get raw bytes.
+     *
+     * @return string
+     */
+    public function getRawKeyMaterial() : string
```

Comment:
Why are you using "getRawKeyMaterial" as a method name here while in the CipherText class you are calling it "getRawBytes"? Both methods look like doing the same, yet have a different name.

I have no clue either what I will be getting when I retrieve "material" other then that it's a string.